### DISCUSSION TITLE
SEQURE S60P

### DISCUSSION THREAD

**Discussion by discip** [Ralim/IronOS#1806] [General]
@Ralim
Today I got the **`S60P`** and decided to share some photos of its PCB:
Maybe this could be helpful.

As not everything in every photo is legible I decided to upload all I took, just in case.

![IMG_20230821_181637](https://github.com/Ralim/IronOS/assets/53649486/a55d89ff-d47a-4f87-895c-3fa1e7571d31)
![IMG_20230821_181704](https://github.com/Ralim/IronOS/assets/53649486/7931675d-a261-474f-87b7-da9943661f7a)
![IMG_20230821_181722](https://github.com/Ralim/IronOS/assets/53649486/e6da6d3e-2cc1-456e-a3c7-d867565633d3)
![IMG_20230821_192806](https://github.com/Ralim/IronOS/assets/53649486/56f41a75-c63e-41e2-90fc-158df47147dd)
![IMG_20230821_193258](https://github.com/Ralim/IronOS/assets/53649486/f3ba76e2-a8da-4f9e-b6b3-4a8a952fe7fd)
![IMG_20230821_193308](https://github.com/Ralim/IronOS/assets/53649486/1bd688fd-4e2f-4058-afe9-d8790559b346)
![IMG_20230821_193432](https://github.com/Ralim/IronOS/assets/53649486/e3e81e26-f8de-4969-99ed-23db3cf3f883)
![IMG_20230821_193442](https://github.com/Ralim/IronOS/assets/53649486/077e3f05-f3e6-42eb-8cb6-ebac3c90a17e)
![IMG_20230821_193514](https://github.com/Ralim/IronOS/assets/53649486/480913a5-89c1-44f6-ada0-f7f6fdc83b41)

**Comment by Ralim** [comment]
The big thing to do would be to compare part-for-part to a S60 and see what they actually changed. Hopefully its just a different USB-PD interface.

If you have a SWD debugger you could try and read out the code with the DVGC testpoints but highly likely it will be locked.

**Reply by discip** [reply]
That's a shame because unfortunately I don't own such a device. 😞

**Reply by discip** [reply]
@Ralim
Today I revisited this post and remembered, that I was not absolutely sure if the thing I bought back in the day of bricking my **`TS80P`** was not actually the device you referred to: [ST-Link+V2](https://www.amazon.com/TELESKY-ST-Link-Editor-Emulator-Downloader/dp/B08SC46QX5/ref=sr_1_2_sspa?crid=35UO6GZQK4UJ6&keywords=ICQUANZX+ST-Link+V2+STM8+STM32&qid=1697451975&sprefix=icquanzx+st-link+v2+stm8+stm32%2Caps%2C192&sr=8-2-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1)

After remembering you answered here I deleted the post from the discussion on this topic.
Meanwhile you answered in the discussion.

So do you think the device I have won't help in the end than?

**Reply by Ralim** [reply]
😁 
So yeah that is exactly the tool to use .

If I was to have a guess of those test pads, its likely they are:

D = SWDIO
V = 3.3V (Just dont connect, and power iron from usb)
G = GND
C = SWCLK

Note there is always a small risk when you connect if the software tries to do an unlock and "blanks" the device.
I have a backup of the S60 bootloader which we could use for rescue, but I'd suggest waiting another day or two and see if sequre answer me (they have started emailing again so I have hope).

**Reply by discip** [reply]
I'm definitely going with the wait! 😁

**Reply by damex** [reply]
looks like it is just a different usb pd interface. I poked around and everything else seem to be the same

![IMG_0796](https://github.com/Ralim/IronOS/assets/450390/bd5ca63b-386c-4fc1-9be2-37e8b551dd56)
![IMG_0797](https://github.com/Ralim/IronOS/assets/450390/857a1f4b-1951-465b-8c7b-38463a2e9fb6)
![IMG_0798](https://github.com/Ralim/IronOS/assets/450390/713f0230-2954-437c-9a6d-ee0d5e61de79)
![IMG_0799](https://github.com/Ralim/IronOS/assets/450390/f136f967-20f2-4eca-987b-747bae5cfff6)
![IMG_0800](https://github.com/Ralim/IronOS/assets/450390/25aef335-156e-43a5-b962-b49535cbe347)
![IMG_0801](https://github.com/Ralim/IronOS/assets/450390/d6b5a4a4-2901-44f6-a8ba-4212702d6328)
![IMG_0802](https://github.com/Ralim/IronOS/assets/450390/e8392e8b-5947-47d7-9855-26ad3cd0b211)
![IMG_0803](https://github.com/Ralim/IronOS/assets/450390/2fb4a300-1ee7-40f9-b9d3-ad1e45e29ad6)

**Reply by damex** [reply]
> 😁 So yeah that is exactly the tool to use .
> 
> If I was to have a guess of those test pads, its likely they are:
> 
> D = SWDIO V = 3.3V (Just dont connect, and power iron from usb) G = GND C = SWCLK
> 
> Note there is always a small risk when you connect if the software tries to do an unlock and "blanks" the device. I have a backup of the S60 bootloader which we could use for rescue, but I'd suggest waiting another day or two and see if sequre answer me (they have started emailing again so I have hope).

it tried connecting DIO/CLK/GND, hold either of the buttons and power board via usb. it boots either normally or boots to sequre 'bootloader' and does not get detected by stlink.

previously i did power s60 (not s60p!) by stlink adapter and shorted 43 and 44 pins to get it detected but here you say not to do that to s60p

**Reply by discip** [reply]
@damex
Thank you very much for the side by side comparison! 👍🏻

@Ralim are you able to tell if it would be safe to flash **`S60P`** with **`IronOS`** intended for the **`S60`**?

I really would like to test those new icons in person! 😀
btw: Could you have a look at https://github.com/Ralim/IronOS/pull/1819 to tell what needs to be changed to make this work? 😁

thanks in advance

**Reply by Ralim** [reply]
Unsure if its safe, I have an S60P on the way to me to tear open and have a look at as sequre hasn't answered my emails again.

**Reply by ArturoGuerra** [reply]
From the surface, it looks like the only change is the USB-PD interface the S60P is using the [FS2711Q ](http://43.138.174.208/wp-content/uploads/2023/02/%E9%80%9F%E8%8A%AF%E5%BE%AE%E7%94%B5%E5%AD%90FS2711P%E6%95%B0%E6%8D%AE%E6%89%8B%E5%86%8C1.1-2022.02.pdf) and from my testing, it looks like the I2C bus is running at 400Khz as it has 4.7k pull up resistors on SDA and SCL

**Reply by Ralim** [reply]
The S60P has arrived this morning.
Aim to have a look at it this weekend.

Loose plan of attack is:

1) Get IronOS-DFU on there to backup their bootloader
2) Get IronOS booting (if any changes required)
3) Figure out details on the PD termination IC

**Reply by maggo1404** [reply]
At the vendor Website found a [Applications notes](http://43.138.174.208/wp-content/uploads/2022/05/FS2711IC-PD-PPS-en..pdf) and [example Code ](http://43.138.174.208/wp-content/uploads/2023/02/FS2711_DEMO_V2_0_01.zip)

**Reply by ArturoGuerra** [reply]
Was able to flash iron-dfu, has to change the vector table offset as it starts at 0x5000 ![image](https://github.com/Ralim/IronOS/assets/18218744/4984be85-b374-4ab4-8b76-e3d29702fc6d)

**Reply by ArturoGuerra** [reply]
![image](https://github.com/Ralim/IronOS/assets/18218744/b736e4e5-a2a5-4f70-bcdc-de354b3cc810)

**Reply by ArturoGuerra** [reply]
@Ralim looking at the code for the S60 it looks like it's using a completely different mechanism for USB-PD as it doesn't use the fusb302 driver, is the plan for the S60P to implement the driver in a similar way? cuz I feel that adding some sort of standardized interface for all power drivers would be better in the future

**Reply by Ralim** [reply]
Yeah the S60 is much more similar to the S60P interface.
So for the S60P it will just implement the simple interface and I'll likely refactor S60 IC to have similar/same interface.

I would love to make all power interfaces standardised, but I _don't have time_ is the honest reason why they are not as refactored as I would like. If you wanted to have a crack and it and open a PR I'd love it though.

**Reply by discip** [reply]
@ArturoGuerra
How did you manage to flash that bootloader?
I just bricked my **`S60P`** trying to do so. 😢

**Reply by ArturoGuerra** [reply]
Had to change the vector table offset to 0x5000 and changed the rom memory region origin to 0x08005000, also I only flashed the runtime to backup the original bootloader never tried to flash the bootloader itself

**Reply by discip** [reply]
@Ralim
What software do you use and what OS are you running to flash?

**Reply by Ralim** [reply]
To recover after trying to flash the bootloader and it broke (as noted in the IronOS-DFU notes I updated):

It depends on what hardware you use as to the easiest way.

ST-Link (V2/V3/Clone):

Windows : ST-Link GUI (From ST)
Linux/Mac `stlink-gui` from the [st-link community package](https://github.com/stlink-org/stlink)
Once you connect to the device, it should warn that flash lock bits are set, select to clear these. Then you can load the .hex or .bin file for either IronOS-DFU or the backup and flash that back.

Segger Jlink:

JFlashLite works well for me, just tell it the chip is an STM32F103.

I have a blackmagic probe somewhere here but havent tried it.

For me I found that getting SWD to connect was rough, so I shorted the reset pin to ground by inserting a thin blade in where the blue arrow is (remember to take note of where the dot is for orientation). Then hit connect and it worked fine to connect-under-reset.
![image](https://github.com/Ralim/IronOS/assets/5425387/18c585a5-38b4-4b3f-9d49-a1abc6edd0ee)

**Reply by ArturoGuerra** [reply]
Started working on adding support for the power IC should hopefully have something working this weekend

**Reply by ArturoGuerra** [reply]
Finished implementing a rough version of the fs2711 just need to test it, I'll make a pull request when I have it working properly

**Reply by damex** [reply]
> Finished implementing a rough version of the fs2711 just need to test it, I'll make a pull request when I have it working properly

s60p without ironos is useless to me and i have s60p with ironos-dfu installed and can help testing ironos on s60p if needed. please tag me in pull request (whenever it is ready) ;)

**Reply by Ralim** [reply]
`dev` branch builds are already partly usable. Though screen is limited.
#1888 is working on better support for the screen.

**Reply by damex** [reply]
@Ralim do i need to do something special to make it work?

iron is on ironos-dfu.
i boot it with ironos dfu and flash last ironos build fot s60p (dev branch artefact) but when iron is power cycled - it always boot in to bootloader.

```
dfu-util -D ~/Downloads/S60P/S60P_EN.dfu
dfu-util 0.11

Copyright 2005-2009 Weston Schmidt, Harald Welte and OpenMoko Inc.
Copyright 2010-2021 Tormod Volden and Stefan Schmidt
This program is Free Software and has ABSOLUTELY NO WARRANTY
Please report bugs to http://sourceforge.net/p/dfu-util/tickets/

Match vendor ID from file: 1209
Match product ID from file: db42
Multiple alternate interfaces for DfuSe file
Opening DFU capable USB device...
Device ID 28e9:0189
Device DFU version 011a
Claiming USB DFU Interface...
Setting Alternate Interface #0 ...
Determining device status...
DFU state(2) = dfuIDLE, status(0) = No error condition is present
DFU mode device DFU version 011a
Device returned transfer size 1024
DfuSe interface name: "Internal Flash "
File contains 1 DFU images
Parsing DFU image 1
Target name: ST...
Image for alternate setting 0, (1 elements, total size = 33972)
Setting Alternate Interface #0 ...
Parsing element 1, address = 0x08005000, size = 33964
Erase   	[=========================] 100%        33964 bytes
Erase    done.
Download	[=========================] 100%        33964 bytes
Download done.
Done parsing DfuSe file
```

**Reply by discip** [reply]
@damex
That's exactly what I reported here (https://github.com/Ralim/IronOS-dfu/issues/24#issuecomment-1962014303) already.

**Reply by damex** [reply]
thanks, tested with new s60p fixes and not only new ironos-dfu build is installed successfully but i could also flash ironos and use s60p as intended.

**Comment by amckinlay** [comment]
What's the point of the S60P? Doesn't the S60 have enough power to satisfy the maximum wattage rating of any C210 tip?

**Reply by Duckle29** [reply]
12V 5A isn't a valid USB-PD voltage or current, so it's kinda difficult to find a power supply that'll be happy with the iron.

![image](https://github.com/Ralim/IronOS/assets/2756925/b7170f5f-1ac7-4eed-bfc3-1c46792279de)

**Comment by discip** [comment]
@Ralim
Today I came across the Sequre firmware page for the **`S60P`** and noticed that a **[hardware change needs to be made for irons of versions PCB_V1.1 and V1.0](https://sequremall.com/blogs/sequre-product-firmware-upgrade/s60p-firmware-upgrade)**.
Do we need to do that as well?

**[The S99 (PCB_V1.4 and V1.3) is also affected.](https://sequremall.com/blogs/sequre-product-firmware-upgrade/s99-soldering-iron-firmware-upgrade)**

![image](https://github.com/Ralim/IronOS/assets/53649486/21552405-9124-4d63-9fc6-0ce5cc0784f6)

**Reply by Ralim** [reply]
Not required, not certain what it's for but if it's on the main rail it would make sense as their firmware resets for me on a bunch of power supplies. 

I haven't done it and will try to maintain functionality without doing the mood.

**Reply by discip** [reply]
So if I do the mod, will my iron still be fully functional? Running IronOS of course. 😃

**Reply by Ralim** [reply]
Yep I believe it won't make anything worse. Might fix some stability issues.

**Reply by discip** [reply]
Great!
Thank you very much! 👍
